lib/fuse_lowlevel.c

```diff
@@ -2772,6 +2788,9 @@ int fuse_session_receive_buf_int(struct fuse_session *se, struct fuse_buf *buf,
 	if (se->conn.proto_minor < 14 || !(se->conn.want & FUSE_CAP_SPLICE_READ))
 		goto fallback;
 
+	if (se->io != NULL && se->io->splice_receive == NULL)
+		goto fallback;
```

Comment:
Can you already ensure this in fuse_lowlevel.c do_init()? So that it does not need to check here all the time?

Reply:
You're right. I'll remove these checks for both read/write and handle it by setting the `conn->capable` field accordingly in `do_init()`